### PR TITLE
Merge LayoutView into ItemView.

### LINKED ISSUE #2251: LayoutView => ItemView PR path
:skull: Farewell, LayoutView!

There are two PRs that need to happen for the LayoutView to be merged into ItemView

- [x] 1. Rename Marionette.View to Marionette.AbstractView #2252
- [x] 2. Migrate LayoutView code into Marionette.ItemView #2253

### PR DESCRIPTION
**Pull request by jmeas**
Resolves #2251

### CODE REVIEW

**Comment by jmeas** [comment]
If anyone wants to investigate, there are two failing tests, likely failing for the exact same reason.

The two tests are [these two.](https://github.com/marionettejs/backbone.marionette/blob/major/test/unit/collection-view.spec.js#L200-L212) Each fail on the second assertion within each test; i.e. [this](https://github.com/marionettejs/backbone.marionette/blob/major/test/unit/collection-view.spec.js#L204). The reason for failure is max call stack size exceeded.

File: src/item-view.js [hunk 1]
```diff
@@ -105,5 +122,122 @@ Marionette.ItemView = Marionette.AbstractView.extend({
...
+    var regions;
+    this._initRegionManager();
+
+    regions = Marionette._getValue(this.regions, this, [options]) || {};
+
+    // Enable users to define `regions` as instance options.
+    var regionOptions = this.getOption.call(options, 'regions');
+
+    // enable region options to be a function
+    regionOptions = Marionette._getValue(regionOptions, this, [options]);
```
Comment (samccone) [inline src/item-view.js]: why [ ]

File: src/item-view.js [hunk 2]
```diff
@@ -105,5 +122,122 @@ Marionette.ItemView = Marionette.AbstractView.extend({
...
+    this.regionManager.invoke('reset');
+  },
+
+  // Internal method to initialize the region manager
+  // and all regions in it
+  _initRegionManager: function() {
+    this.regionManager = this.getRegionManager();
+    this.regionManager._parent = this;
+
+    this.listenTo(this.regionManager, 'before:add:region', function(name) {
```
Comment (samccone) [inline src/item-view.js]: you know what would be nice....

```js
Marionette.listenTo(this, [
  {'before:add:region': function(){}},
  {'add:region': function(){}}]
});
```

Comment (thejameskyle) [inline src/item-view.js]: wtf is going on with that

Comment (samccone) [inline src/item-view.js]: A way to setup multiple listenTo's in a declarative fashion.

Comment (thejameskyle) [inline src/item-view.js]: Do you even Backbone.Events bro
```js
Events.listenTo(this, {
  'before:add:region': function() {},
  'add:region': function() {}
});
```

Comment (samccone) [inline src/item-view.js]: lol nice

Comment (samccone) [inline src/item-view.js]: however... my syntax allows for duplicate event  registration

Comment (thejameskyle) [inline src/item-view.js]: stop it

Comment (samccone) [inline src/item-view.js]: wat it is true

Comment (thejameskyle) [inline src/item-view.js]: go home

**Comment by jmeas** [comment]
From the perspective of **code**, this can be merged. I would be fine if we merged it now so that other PRs in flight can be merged without conflict. I can update the rest of the docs separately.

Thoughts @marionettejs/marionette-core ?

File: src/item-view.js (hunk 1 above)
Comment (thejameskyle) [inline src/item-view.js]: @samccone https://github.com/marionettejs/backbone.marionette/pull/2253/files#diff-8a756280cd88a053efb536867eafdd90L99

Comment (jmeas) [inline src/item-view.js]: Let's talk about refactors in a separate issue. This is a direct port of the LayoutView into the ItemView; no code changes.

Comment (jasonLaster) [inline src/item-view.js]: Yup

File: src/item-view.js (hunk 2 above)
Comment (jasonLaster) [inline src/item-view.js]: Lolz

**Comment by jasonLaster** [comment]
Wow. It went in without a hitch. 

One thing is that you'll want to rebase major to get the shortcut methods. 

Also doc merging can wait.  And this makes me think the region mixin idea is a good one too.

**Comment by jmeas** [comment]
> One thing is that you'll want to rebase major to get the shortcut methods.

Will-do.

**Comment by jmeas** [comment]
Major was rebased. Updated with your shortcut methods @jasonLaster and @megawac's `destroyImmediate`

**Comment by thejameskyle** [comment]
:+1:

**Comment by jmeas** [comment]
WOOOP

**Pull request merged**